OpenSimRT/RealTime/include/RealTimeAnalysis.h

```diff
@@ -0,0 +1,234 @@
+ * capture system and the InverseKinematics input. Termination of the
+ * acquisition should be handled by throwing an exception.
+ */
+typedef std::function<MotionCaptureInput()> DataAcquisitionFunction;
+
+/**
+ * @brief Provides a convinient interface for performing RT musculoskeletal
+ * analysis. It creates one thread for the data acquisition, IK and filtering,
+ * and one processing thread for the rest the analysis (ID, SO and JR).
+ */
+class RealTime_API RealTimeAnalysis {
+ public:
+    struct UnfilteredData {
+        double t;
+        SimTK::Vector q;
+        std::vector<ExternalWrench::Input> externalWrenches;
+
+        /**
+         * Converts the UnfilteredData struct fields into a single SimTK::Vector
+         */
+        SimTK::Vector toVector();
+
+        /**
+         * Returns the size of the UnfilteredData fields in number of
+         * sizeof(double).
+         */
+        int size();
+    };
+
+    struct FilteredData {
+        double t;
+        SimTK::Vector q;
+        SimTK::Vector qd;
+        SimTK::Vector qdd;
+        std::vector<ExternalWrench::Input> externalWrenches;
+
+        /**
+         * Assigns the struct fields of FilteredData from SimTK::Vectors. The
+         * first nq elements of the input SimTK::Vectors contain the filtered
+         * generalized coordinates (and derivatives), and the rest elements
+         * correspond to the filtered wrench data.
+         */
+        void fromVector(const double& time, const SimTK::Vector& x,
+                        const SimTK::Vector& xd, const SimTK::Vector& xdd,
+                        const int& nq);
+    };
+
+    struct Output {
+        // filtered IK
+        double t;
+        SimTK::Vector q;
+        SimTK::Vector qd;
+        SimTK::Vector qdd;
+        SimTK::Vector grfRightWrench;
+        SimTK::Vector grfLeftWrench;
+
+        // ID
+        SimTK::Vector tau;
+
+        // SO
+        SimTK::Vector am;
+        SimTK::Vector fm;
+        SimTK::Vector residuals;
+
+        // JRA
+        SimTK::Vector_<SimTK::SpatialVec> reactionWrenches;
+        SimTK::Vector reactionWrenchVector; // alternative representation
+    };
+
+    struct Parameters {
+        // acquisition function
+        DataAcquisitionFunction dataAcquisitionFunction;
+
+        // lp smooth filter parameters
+        LowPassSmoothFilter::Parameters filterParameters;
+
+        // ik parameters
+        std::vector<InverseKinematics::MarkerTask> ikMarkerTasks;
+        std::vector<InverseKinematics::IMUTask> ikIMUTasks;
+        double ikConstraintsWeight;
+        double ikAccuracy;
+
+        // id + jr parameters
+        std::vector<ExternalWrench::Parameters> wrenchParameters;
+
+        // so parameters
+        bool solveMuscleOptimization;
+        MuscleOptimization::OptimizationParameters muscleOptimizationParameters;
+        MomentArmFunctionT momentArmFunction;
+    };
+
+    struct Loggers {
```

Comment:
Do we need the loggers here?

Reply:
I have restricted the access to the individual modules. So logger initialization can't happen from the user code. I thought it might be more compact if the user code only calls `pipeline.initializeLoggers()` once.